### DISCUSSION TITLE
how to build post command using custom target when after done `executable` command.

### DISCUSSION THREAD

**Discussion by Nell69Rock** [mesonbuild/meson#10954] [Q&A]
In most situations, the custom target is executed first. before executing the executable command.

assume that below command

executable('foo.elf', bar.c)

custom_target('foo.dump, 'foo.elf', @DUMP_CMD@)  #always running this line first!!
ninja: error: '/USR/PATH/foo.elf', needed by 'foo.dump', missing and no known rule to make it

**Comment by eli-schwartz** [comment]
That is not valid custom_target syntax.

Nevertheless, the documentation at https://mesonbuild.com/Reference-manual_functions.html#custom_target states that you can pass the return value of `foo_elf = executable('foo.elf', 'bar.c')` to a custom_target command. Simply use it via the returned variable, not as a string.
